android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginActivity.kt

```diff
@@ -42,9 +44,18 @@ class LoginActivity :
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    loginService.loginActivity = this
+    loginService.runningActivity = this
     loginViewModel.apply {
-      navigateToHome.observe(this@LoginActivity, { loginService.navigateToHome() })
+      navigateToHome.observe(
+        this@LoginActivity,
+        {
+          if (loginViewModel.loginViewConfiguration.value?.enablePin == true) {
```

Comment:
There should be a check here if user has previously gone through the set-up pin process and have a pin already.

Reply:
Hey @RaaziaTarique , 
as per tickets description #912  and #911 **"This selection screen will be shown immediately after the initial username/password login"**

this is also discussed and demo with @dubdabasoduba too, app need to setup PIN every time user login with username/password, if user forgot the pin, it can be reset with this flow.

Reply:
Alright, got it.